packages/shell-api/src/collection.spec.ts

```diff
@@ -1100,24 +1100,26 @@ describe('Collection', () => {
 
       beforeEach(() => {
         mockResult = { value: {} };
-        serviceProvider.findAndModify.resolves(mockResult);
+        serviceProvider.findOneAndUpdate.resolves(mockResult);
+        serviceProvider.findOneAndReplace.resolves(mockResult);
+        serviceProvider.findOneAndDelete.resolves(mockResult);
       });
 
-      it('returns result.value from serviceProvider.findAndModify', async() => {
-        expect(await collection.findAndModify({ query: {} })).to.equal(mockResult.value);
+      it('returns result.value from serviceProvider.findOneAndReplace', async() => {
```

Comment:
IIUC how the mock is set up in `beforeEach` because all three methods are mocked with one shared `mockResult` this test doesn't guarantee that the `result.value` is returned from `findOneAndReplace` right? Maybe it would be better to have separate mocks there or just `spy` on the method and asset that it's being called, what do you think?

Reply:
Yeah, good point – will do that in an update PR :)